module/input/Camera/src/Camera.cpp

```diff
@@ -1,129 +1,364 @@
+                        message      = set_setting(setting, value);
+                    }
+
+                    // Enumeration setting
+                    else if (ARV_IS_GC_ENUMERATION(feature)) {
+                        auto setting      = reinterpret_cast<ArvGcEnumeration*>(feature);
+                        std::string value = cfg.second.as<std::string>();
+                        message           = set_setting(setting, value);
+                    }
                     else {
-                        resetAravisCamera(cam, config);
+                        log<NUClear::ERROR>(fmt::format("The type of setting {} is not currently handled", key));
                     }
-                }
 
-                else {
-                    log<NUClear::FATAL>("Unsupported camera driver:", driver);
+                    // If we had a message pass it onto the user
+                    if (!message.empty()) {
+                        log<NUClear::INFO>(fmt::format("Updated {} on {} camera: {}", key, name, message));
+                    }
                 }
+                catch (const std::runtime_error& e) {
+                    log<NUClear::ERROR>(fmt::format("Error setting {} on {} camera: {}", key, name, e.what()));
+                }
+            }
+
+            // Check for gigevision device and set packet size to jumbo packets
+            if (arv_camera_is_gv_device(cam.get())) {
+                arv_camera_gv_set_packet_size(cam.get(), 8192);
+                g_object_set(stream.get(), "packet-resend", ARV_GV_STREAM_PACKET_RESEND_NEVER, nullptr);
+            }
+
+            // Add buffers to the queue
+            int payload_size = arv_camera_get_payload(cam.get());
+            for (size_t i = 0; i < config["buffer_count"].as<size_t>(); i++) {
+                // TODO(trent) Eventually we should use preallocated page aligned data so that we can map directly to
+                // the GPU.
+                // TODO(trent) Make an std::vector and use it in the buffer to avoid copying to one later
+                arv_stream_push_buffer(stream.get(), arv_buffer_new(payload_size, nullptr));
+            }
+
+            // Connect signal events
+            g_signal_connect(stream.get(), "new-buffer", G_CALLBACK(&Camera::emit_image), &it->second);
+            g_signal_connect(
+                arv_camera_get_device(cam.get()), "control-lost", G_CALLBACK(&Camera::control_lost), nullptr);
+            // Start aquisition
+            arv_camera_start_acquisition(cam.get());
+            arv_stream_set_emit_signals(stream.get(), true);
+        });
+
+        on<Shutdown>().then([this] {
+            for (auto& camera : cameras) {
+                // Stop the video stream.
+                arv_camera_stop_acquisition(camera.second.camera.get());
+                arv_stream_set_emit_signals(camera.second.stream.get(), false);
             }
+            arv_debug_shutdown();
```

Comment:
is this needed?

Reply:
maybe wrap this in a config param (along with the thing that enables debugging) so we can enable it if needed

Reply:
Removed it, it's not very useful anyway